addons/html_builder/static/src/plugins/add_element_option.js

```diff
@@ -1,22 +1,179 @@
+import { Plugin } from "@html_editor/plugin";
+import { registry } from "@web/core/registry";
+import { _t } from "@web/core/l10n/translation";
+import { resizeGrid, setElementToMaxZindex } from "@html_builder/utils/grid_layout_utils";
+import { withSequence } from "@html_editor/utils/resource";
 import { Component } from "@odoo/owl";
-import { defaultBuilderComponents } from "../core/default_builder_components";
-import { Button } from "./button";
+import { defaultBuilderComponents } from "@html_builder/core/default_builder_components";
 
 export class AddElementOption extends Component {
     static template = "html_builder.AddElementOption";
-    static components = {
-        ...defaultBuilderComponents,
-        Button,
+    static components = defaultBuilderComponents;
+    static props = {
+        level: { type: String, optional: true },
+        getLevel: { type: Function, optional: true },
+    };
+    static defaultProps = {
+        level: "0",
+    };
+    setup() {
+        if (this.props.getLevel) {
+            const editingElement = this.env.getEditingElement();
+            this.props.level = this.props.getLevel(editingElement);
+        }
+    }
+}
+export class AddElementOptionPlugin extends Plugin {
+    static id = "AddElementOption";
+    static dependencies = ["history", "media"];
+    resources = {
+        builder_options: [
+            withSequence(11,
+                {
+                    OptionComponent: AddElementOption,
+                    selector: ".s_masonry_block, .s_numbers_grid",
```

Comment:
Why not use AddElementOption directly in the layoutOption and mandatory Option ?